rpc_proxy.go

```diff
@@ -44,8 +48,20 @@ var (
 		defaultServerAddr,
 		"the host:port address which the HTTP proxy should listen on",
 	)
+
+	serviceStateMap = map[serviceState]string{
+		statusServing:      "SERVING",
+		statusNotServing:   "NOT_SERVING",
```

Comment:
when this is filled?

Reply:
this is actually never sent by the proxy back to the client. The proxy should be considered in NOT_SERVING state if the call to /healtcheck fails for whatever reason.